staging/src/k8s.io/kubectl/pkg/cmd/drain/drain.go

```diff
@@ -297,34 +298,45 @@ func (o *DrainCmdOptions) RunDrain() error {
 	}
 
 	drainedNodes := sets.NewString()
-	var fatal error
+	var deleteOrEvictErrors []error
+	var deleteOrEvictResultMutex sync.Mutex
+	var deleteOrEvictWaitGroup sync.WaitGroup
+	deleteOrEvictWaitGroup.Add(len(o.nodeInfos))
 
 	for _, info := range o.nodeInfos {
-		if err := o.deleteOrEvictPodsSimple(info); err == nil {
-			drainedNodes.Insert(info.Name)
-			printObj(info.Object, o.Out)
-		} else {
-			fmt.Fprintf(o.ErrOut, "error: unable to drain node %q, aborting command...\n\n", info.Name)
-			remainingNodes := []string{}
-			fatal = err
-			for _, remainingInfo := range o.nodeInfos {
-				if drainedNodes.Has(remainingInfo.Name) {
-					continue
-				}
-				remainingNodes = append(remainingNodes, remainingInfo.Name)
+		go func(info *resource.Info) {
+			err := o.deleteOrEvictPodsSimple(info)
+			deleteOrEvictResultMutex.Lock()
+			defer deleteOrEvictResultMutex.Unlock()
+			defer deleteOrEvictWaitGroup.Done()
+
+			if err == nil {
+				drainedNodes.Insert(info.Name)
+				printObj(info.Object, o.Out)
+			} else {
+				fmt.Fprintf(o.ErrOut, "error: unable to drain node %q\n\n", info.Name)
+				deleteOrEvictErrors = append(deleteOrEvictErrors, err)
 			}
+		}(info)
+	}
 
-			if len(remainingNodes) > 0 {
-				fmt.Fprintf(o.ErrOut, "There are pending nodes to be drained:\n")
-				for _, nodeName := range remainingNodes {
-					fmt.Fprintf(o.ErrOut, " %s\n", nodeName)
-				}
-			}
-			break
+	deleteOrEvictWaitGroup.Wait()
+	remainingNodes := []string{}
+	for _, remainingInfo := range o.nodeInfos {
```

Comment:
In light of the opt-in flag for parallel drain that I just pushed I think we still need this printing for the serial case. This will also print the list of nodes that failed to drain in the parallel case in a single line in the bottom. As you mentioned in your other comment the scrolling failure messages are easy to lose, so I think this final output is helpful in the parallel case too

Reply:
I agree with Michael here, if this is not needed for parallel case, move this to the serial one, inside the `if-else` condition above.

Reply:
The "are we parallel?" conditional above is actually being invoked in a loop over nodes, and this printing needs to happen only once. I can add another conditional on the same flag here though if you'd like?

I do think the absence of this output will have a negative impact on the UX in the parallel case but if you disagree I'm fine being overruled